care/facility/models/asset.py

```diff
@@ -54,6 +55,15 @@ class RoomType(enum.Enum):
         null=True, blank=True, default=None, max_length=200
     )
 
+    def delete(self, *args):
+        if UserDefaultAssetLocation.objects.filter(location=self).exists():
+            error = f"Cannot delete AssetLocation {self} because they are referenced as `location` in UserDefaultAssetLocation records."
```

Comment:
It might make more sense to create a static errors and make it translatable as well, also keep the errors very human readable like